src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -442,6 +444,63 @@ void AggregatingStep::transformPipeline(QueryPipelineBuilder & pipeline, const B
         return;
     }
 
+    std::optional<std::string> changed_key;
+    std::optional<DataTypePtr> key_data_type;
+    if (transform_params->params.keys.size() == 1)
+    {
+        const auto & key = transform_params->params.keys.front();
```

Comment:
don't forget to check that `key` is not among aggregate functions params

Reply:
I've just checked, and it seems that in my current implementation, having key as aggregate function param does not ruin anything. Why could it?

Reply:
if we do not remove original column and it still comes to aggregator then we probably good